app/src/main/java/com/mdg/droiders/samagra/shush/AlarmScheduler.java

```diff
@@ -30,52 +36,299 @@ public AlarmScheduler(Context context) {
     }
 
     /**
+     * Sets an alarm to shush the phone for a single day.
+     *
+     * @param startTime The time at which phone is to be shushed
+     * @param endTime   The time at which phone is to be un-shushed
+     * @param day       int type for week days starting from monday and zero indexed.
+     *                  For e.g. - Monday is 0, Tuesday is 1, ... and so on
+     * @param rowID     Unique primary key of the shush alarm row.
+     */
+    public void setSingleDayAlarm(Calendar startTime, Calendar endTime, int day, Integer rowID) {
+        if (day > 6 || rowID == null) {
+            return;
```

Comment:
No exception?

Reply:
If the method throws and exception, then I will have to catch it every time I use the method. I won't be passing anything that should cause an exception while using this method.

Reply:
Isn't this a wrong way? You won't pass such values intentionally, but wouldn't it be better to catch that exception and show the user an error message or something? 

[This](https://stackoverflow.com/questions/18679090/when-to-catch-the-exception-vs-when-to-throw-the-exceptions#18679131) explains why there should be an exception. If by chance something happens and condition `day > 6` or `rowID == null` is evaluated to true then the user wouldn't know that there is some problem. He wouldn't know if his action set an alarm or not.